src/model/stock_predictor_manager.py

```diff
@@ -0,0 +1,218 @@
+import datetime
+import sys
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+import torch
+from sklearn.preprocessing import MinMaxScaler
+from torch import nn, optim
+from torch.utils.data import DataLoader, TensorDataset
+
+from ..server.constants import QUERIES
+from ..server.database import DatabaseProvider
+from .stock_predictor import StockPredictor
+from .stock_predictor_config import StockPredictorConfig
+
+
+class StockPredictorManager:
+    """Class for managing StockPredictor GRU model"""
+
+    def __init__(self) -> None:
+        self.config = StockPredictorConfig()
+
+        self._initialize_model()
+
+        self.train_size_pct: float = 0.8
+        self.scaler: MinMaxScaler = MinMaxScaler()
+
+        self.criterion: nn.MSELoss = nn.MSELoss()
+        self.optimizer: optim.adam.Adam = optim.Adam(  # pylint: disable=no-member
+            self.stock_predictor.parameters(), lr=self.config.learning_rate
+        )
+
+    def _initialize_model(self) -> None:
+        """Initializing model with parameters from config dataclass."""
+        self.stock_predictor: StockPredictor = StockPredictor(
+            self.config.input_dim,
+            self.config.hidden_dim,
+            self.config.num_layers,
+            self.config.output_dim,
+        )
+
+    def _get_data(self) -> pd.DataFrame:
+        """Return all data from the stock values table."""
+        data: list = []
+
+        with DatabaseProvider.handler() as handler:
+            handler().execute(QUERIES.SELECT_ALL_RATE_HISTORY)
+            data = handler().fetchall()
+            data = [{"date": date.strftime("%Y-%m-%d"), "avg": avg} for date, avg in data]
+
+        data_frame = pd.DataFrame(data, columns=["date", "value"])
```

Comment:
Jako że używamy torch'a, dlaczego nie został użyty jego natywny format (Tensor)?

Reply:
Converting directly to the Tensor data type instead of using Pandas as an intermediate type is a good idea. Currently, testing every change related to retrieving data from a database is very difficult. I would suggest pushing the changes to the main branch, integrating the model with the backend, and then removing the use of the Pandas library.